hazelcast/test/src/sql_test.cpp

```diff
@@ -1463,6 +1506,57 @@ TEST_F(SqlTest, exception)
                  hazelcast::client::sql::hazelcast_sql_exception);
 }
 
+TEST_F(SqlTest, find_with_page_sync_iterator)
+{
+    create_mapping();
+    auto numbers = populate_map(map, 500);
+
+    auto searchee = numbers[numbers.size() / 2];
+    auto result = select_all();
+
+    auto found_page_itr = std::find_if(
+        result->pbegin(),
+        result->pend(),
+        [searchee](std::shared_ptr<sql::sql_page> page){
+            return find_if(
+                begin(page->rows()),
+                end(page->rows()),
+                [searchee](const sql::sql_page::sql_row& row){
+                    return row.get_object<int>(1).value() == searchee;
+                }
+            ) != end(page->rows());
+        }
+    );
+
+    ASSERT_NE(found_page_itr, result->pend());
+
+    bool exist = any_of(
+        begin(found_page_itr->rows()),
+        end(found_page_itr->rows()),
+        [searchee](const sql::sql_page::sql_row& row){
+            return row.get_object<int>(1).value() == searchee;
+        }
+    );
+
+    ASSERT_TRUE(exist);
+}
+
+TEST_F(SqlTest, timeout_for_page_iterator_sync)
+{
+    // `generate_stream(1)` generates a row per seconds, so it will guaranteed
+    // that it will timeout
+    auto result =
+      client.get_sql().execute("SELECT * FROM TABLE(generate_stream(1))").get();
+
+    auto statement = [&result](){
+        auto it = result->pbegin(std::chrono::milliseconds{ 1 });
+        ++it;
+        ++it;
+    };
+
+    ASSERT_THROW(statement(), hazelcast::client::exception::no_such_element);
```

Comment:
I would prefer a different exception type like timeout as a user of library.

Reply:
By mistaken :)

Fixed.